src/components/UserSelector.tsx

```diff
@@ -22,13 +39,29 @@ export const UserSelector: React.FC = () => {
       </div>
 
       <div className="dropdown-menu" id="dropdown-menu" role="menu">
+
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+          {open ? (
+            users.map(user => {
+              const { id, name } = user;
+
+              return (
+                <a
+                  href={`#user-${id}`}
+                  className="dropdown-item"
+                  key={id}
+                  onClick={() => {
+                    selectUser(user);
+                    handleOpen();
+                  }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here